c/include/libsbp/sbp.h

```diff
@@ -43,19 +43,42 @@ extern "C" {
 
 /** Default sender ID. Intended for messages sent from the host to the device. */
 #define SBP_SENDER_ID 0x42
+/** Header length in bytes. */
+#define SBP_HEADER_LEN 6
+/** CRC length in bytes. */
+#define SBP_CRC_LEN 2
+/** SBP_MSG_ID to use to register frame callback for ALL messages. */
+#define SBP_MSG_ALL 0
 
-/** SBP callback function prototype definition. */
+/** SBP callback function prototype definitions. */
 typedef void (*sbp_msg_callback_t)(u16 sender_id, u8 len, u8 msg[], void *context);
+typedef void (*sbp_frame_callback_t)(u16 sender_id, u16 msg_type,
+                                     u8 payload_len, u8 payload[],
+                                     u16 frame_len, u8 frame[], void *context);
 
+/** SBP callback type enum:
+ * SBP_TYPE_CALLBACK are the original callbacks in libsbp without framing args
+ * SBP_TYPE_FRAME are raw frame callbacks that include framing data as args.
+ * This enum is stored on each sbp_msg_callback_node struct to identify how
+ * to cast the callback function pointer stored within.
+ */
+enum sbp_cb_type{
+  SBP_TYPE_CALLBACK,
+  SBP_FRAME_CALLBACK
```

Comment:
in true C style, we could turn this into a bit field to allow the `sbp_process_frame` function to be super flexible, just throw in a bitmask to select the callback styles you'd like to service. as easy as:
```
    if ((node->cb_type & callback_select_mask) &&
        ((node->msg_type == msg_type) || (node->msg_type == SBP_MSG_ALL))) {
```